fe/fe-core/src/test/java/org/apache/doris/nereids/rules/expression/rules/SimplifyCastRuleTest.java

```diff
@@ -0,0 +1,54 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.nereids.rules.expression.rules;
+
+import org.apache.doris.nereids.rules.expression.ExpressionRewriteTestHelper;
+import org.apache.doris.nereids.rules.expression.ExpressionRuleExecutor;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.Slot;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+class SimplifyCastRuleTest extends ExpressionRewriteTestHelper {
+
+    @Test
+    public void testSimplify() {
+        executor = new ExpressionRuleExecutor(ImmutableList.of(SimplifyCastRule.INSTANCE));
+        assertRewriteAfterSimplify("CAST('1' AS STRING)", "'1'", null);
+        assertRewriteAfterSimplify("CAST('1' AS VARCHAR)", "'1'", null);
+        assertRewriteAfterSimplify("CAST(1 AS DECIMAL)", "1", null);
+        assertRewriteAfterSimplify("CAST(1000 AS DECIMAL)", "1000", null);
+        assertRewriteAfterSimplify("CAST(1 AS DECIMALV3)", "1", null);
+        assertRewriteAfterSimplify("CAST(1000 AS DECIMALV3)", "1000", null);
+    }
+
+    private void assertRewriteAfterSimplify(String expr, String expected, Map<String, Slot> slotNameToSlot) {
+        Expression needRewriteExpression = PARSER.parseExpression(expr);
+        if (slotNameToSlot != null) {
+            needRewriteExpression = replaceUnboundSlot(needRewriteExpression, slotNameToSlot);
+        }
+        Expression rewritten = SimplifyCastRule.INSTANCE.rewrite(needRewriteExpression, context);
+        Expression expectedExpression = PARSER.parseExpression(expected);
+        Assertions.assertEquals(expectedExpression.toSql(), rewritten.toSql());
```

Comment:
i think we should not use to sql to check result, because we need to ensure the rewritten expression has correct type